tests/integration_tests/lightgbm_tuner_tests/test_optimize.py

```diff
@@ -647,6 +647,20 @@ def test_get_best_booster(self) -> None:
         with pytest.raises(ValueError):
             tuner2.get_best_booster()
 
+    @pytest.mark.parametrize("mock_exists, expected", [(False, True), (True, False)])
+    def test_model_dir(self, mock_exists: bool, expected: bool) -> None:
```

Comment:
A nit, but how about using the same variable name in both `test_model_dir`?
```suggestion
    @pytest.mark.parametrize("dir_exists, expected", [(False, True), (True, False)])
    def test_model_dir(self, dir_exists: bool, expected: bool) -> None:
```